herddb-core/src/main/java/herddb/data/consistency/TableDataChecksum.java

```diff
@@ -0,0 +1,112 @@
+/*
+ Licensed to Diennea S.r.l. under one
+ or more contributor license agreements. See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership. Diennea S.r.l. licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing,
+ software distributed under the License is distributed on an
+ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ KIND, either express or implied.  See the License for the
+ specific language governing permissions and limitations
+ under the License.
+
+ */
+package herddb.data.consistency;
+
+import herddb.codec.RecordSerializer;
+import herddb.core.TableSpaceManager;
+import herddb.model.TransactionContext;
+import herddb.model.DataScanner;
+import herddb.model.DataScannerException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import herddb.model.commands.ScanStatement;
+import herddb.utils.DataAccessor;
+import herddb.core.AbstractTableManager;
+import herddb.core.DBManager;
+import herddb.model.Column;
+import herddb.model.Table;
+import herddb.sql.TranslatedQuery;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import net.jpountz.xxhash.StreamingXXHash64;
+import net.jpountz.xxhash.XXHashFactory;
+
+/**
+ * digest creation by scanning the table
+ *
+ * @author Hamado.Dene
+ */
+public abstract class TableDataChecksum {
+
+    private static final Logger LOGGER = Logger.getLogger(TableDataChecksum.class.getName());
+    private static final XXHashFactory factory = XXHashFactory.fastestInstance();
+    //private static final int SEED = 0x9747b28c;
+    private static final int SEED = 0;
```

Comment:
Can you tell us more about you choosing this specific SEED ("0")?

Reply:
My scope is to generate the same hash in the leader and the follower. So I have not generate a random number for the seed.
0 is the default value and with the same values, I should always get the same hash.

I'm not sure is the better approach, do you have some suggestion?

Reply:
No, i have no suggestion